File: Assets/Smash Way/Scripts CS/GameOther/ForceExplosion.cs
using UnityEngine;
using System.Collections;

public class ForceExplosion : MonoBehaviour {

	public GameObject Explosion;
	public GameObject Effector;
	public AudioClip AudioList;
	private bool Stop=false;

	public void OnMouseDown(){
		OnTouchDown();
	}


	void OnTouchDown(){
		if(!Stop){
			Effector.SetActive(false);
			gameObject.GetComponent<Rigidbody>().isKinematic=false;
			Explosion.SetActive(true);
			if(gameObject.GetComponent<Animation>()){gameObject.GetComponent<Animation>().enabled=false;}
			gameObject.GetComponent<AudioSource>().PlayOneShot(AudioList);
			Stop=true;
		}
	}
}

File: Assets/Smash Way/Scripts JS/GameOther/ForceExplosion.js
#pragma strict

var Explosion:GameObject;
var Effector:GameObject;
var AudioList:AudioClip;
private var Stop:boolean=false;

function OnMouseDown(){
OnTouchDown();
}


function OnTouchDown(){
if(!Stop){
Effector.SetActive(false);
gameObject.GetComponent.<Rigidbody>().isKinematic=false;
Explosion.SetActive(true);
if(gameObject.GetComponent.<Animation>()){gameObject.GetComponent.<Animation>().enabled=false;}
gameObject.GetComponent.<AudioSource>().PlayOneShot(AudioList);
Stop=true;
}
}